feat: `loader.createSchema()` (#14759)
withastro/astro: .changeset/fresh-rocks-sing.md

---
'astro': major
---

> Removes the ability for content loaders schemas to be functions and adds a new equivalent `createSchema()` property (Loader API) - ([v6 upgrade guidance](https://deploy-preview-12322--astro-docs-2.netlify.app/en/guides/upgrade-to/v6/TODO:))

Reviewer: ```suggestion
Removes the option to define dynamic schemas in content loaders as functions and adds a new equivalent `createSchema()` property (Loader API) - ([v6 upgrade guidance](https://deploy-preview-12322--astro-docs-2.netlify.app/en/guides/upgrade-to/v6/TODO:))
```
